app/src/main/java/nu/yona/app/ui/LaunchActivity.java

```diff
@@ -118,10 +122,71 @@ else if (!getSharedUserPreferences().getBoolean(PreferenceConstant.STEP_PASSCODE
 		}
 		else if (!TextUtils.isEmpty(getSharedUserPreferences().getString(PreferenceConstant.YONA_PASSCODE, "")))
 		{
-			startNewActivity(bundle, YonaActivity.class);
+			checkForUserEntityVersion();// validate any updates to user entity before moving to next screen.
+		}
+	}
+
+	private void checkForUserEntityVersion()
+	{
+		try
+		{
+			JSONObject userJsonObj = APIManager.getInstance().getAuthenticateManager().getStoredUserObjectFromDB();
+			if (userJsonObj.isNull("version") || userJsonObj.getInt("version") != AppConstant.USER_ENTITY_VERSION)
```

Comment:
With what I said at ``getStoredUserObjectFromDB``, the above two lines can now become:
```
User user = APIManager.getInstance().getAuthenticateManager().getUser();
if (user.getVersion() != AppConstant.USER_ENTITY_VERSION)
```

Reply:
Done.